### ISSUE TITLE
Issue Fixes for Line in BarChart using secondaryYAxis

### ISSUE THREAD

**Issue by sandeep-synclovis** [Abhinandan-Kushwaha/gifted-charts-core#4]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `gifted-charts-core@0.0.21` for the project I'm working on.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/gifted-charts-core/src/BarChart/index.ts b/node_modules/gifted-charts-core/src/BarChart/index.ts
index 76b53ee..4e63116 100644
--- a/node_modules/gifted-charts-core/src/BarChart/index.ts
+++ b/node_modules/gifted-charts-core/src/BarChart/index.ts
@@ -192,7 +192,7 @@ export const useBarChart = (props: extendedBarChartPropsType) => {
     maxAndMin.maxItem
   );
   const secondaryMaxValue = lineConfig.isSecondary
-    ? secondaryMaxAndMin.maxItem
+    ? props.secondaryYAxis.maxValue
     : maxValue;
   const mostNegativeValue = props.mostNegativeValue ?? maxAndMin.minItem;
 
diff --git a/node_modules/gifted-charts-core/src/utils/index.tsx b/node_modules/gifted-charts-core/src/utils/index.tsx
index a2e94a0..5056a83 100644
--- a/node_modules/gifted-charts-core/src/utils/index.tsx
+++ b/node_modules/gifted-charts-core/src/utils/index.tsx
@@ -709,7 +709,7 @@ export const getAxesAndRulesProps = (
   };
   if (
     (props.secondaryYAxis || props.lineConfig?.isSecondary) &&
-    maxValue !== undefined
+    maxValue !== undefined && props.secondaryYAxis.maxValue == undefined
   ) {
     axesAndRulesProps.secondaryYAxis = { ...props.secondaryYAxis, maxValue };
   }
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by Abhinandan-Kushwaha** [comment]
Fixed from version `0.1.12` onwards.

**Comment by Abhinandan-Kushwaha** [comment]
Fixed from version `0.1.12` onwards.

**Comment by sandeep-synclovis** [comment]
Thanks
